backend/src/middlewares/auth/strategies/facebook.strategy.ts

```diff
@@ -0,0 +1,94 @@
+import { ENV } from '@common/enums/app/app';
+import { prisma } from '@data/prisma-client';
+import { verifyToken } from '@helpers/helpers';
+import { JwtPayload } from 'jsonwebtoken';
+import { Strategy as FacebookStrategy } from 'passport-facebook';
+
+const facebookStrategy = new FacebookStrategy(
+  {
+    clientID: ENV.FACEBOOK.APP_ID as string,
+    clientSecret: ENV.FACEBOOK.SECRET_KEY as string,
+    callbackURL: `${ENV.API.V1_PREFIX}/auth/facebook/redirect`,
+    profileFields: [
+      'email',
+      'gender',
+      'displayName',
+      'hometown',
+      'location',
+      'birthday',
+      'picture',
+    ],
+    passReqToCallback: true,
+  },
+  async (req, accessToken, refreshToken, profile, done) => {
+    console.log(profile);
+    console.log(parseInt(profile._json.birthday.substr(-4)));
+
+    let user;
```

Comment:
Why did you declare the untyped `user` variable here?  Couldn't it be declared inside the try-catch?